src/agent/agent.js

```diff
@@ -198,21 +231,35 @@ export class Agent {
 
         // This update loop ensures that each update() is called one at a time, even if it takes longer than the interval
         const INTERVAL = 300;
+        let last = Date.now();
         setTimeout(async () => {
             while (true) {
                 let start = Date.now();
-                await this.bot.modes.update();
+                await this.update(start - last);
```

Comment:
Will this ever not be equal to `INTERVAL`?

Reply:
No, setTimeouts is not 100% reliable. I'd rather calculate it manually to make sure it's accurate